### PR TITLE
Move pack content to top level.

### PR DESCRIPTION
**Pull request by manasdk**
* Following pattern from https://github.com/StackStorm/st2-slack

Thoughts?

### CODE REVIEW

**Comment by jfryman** [comment]
No. I purposely did this because there is a bug with `packs.deploy`.

**Comment by jfryman** [comment]
https://stackstorm.atlassian.net/browse/STORM-1693

**Comment by manasdk** [comment]
I don't buy your bug :p

pack_name = repo_name

**Comment by jfryman** [comment]
I do not think this is the behavior we want.
Consider the pack in question. st2-Kafka. If I were to name it just kafka to
keep the pack_name == git_repo, then this will not allow me to ever fork the
actual Kafka project as there is a namespace collision.
Likewise, there are going to be conflicting patterns and packs at some point in
the future. A world where everyone is going to try to reinvent the same wheel.
In this sense, there will be fryman-kafka and phool-kafka and whatnot.
But, the services I am downloading are not any of these names. It's just Kafka.
In both of the above scenarios, I am developing the Kafka module, and the
metadata will reference it as such. But, because of the current behavior, rules
and aliases will fail to load because st2-kafka != kafka
It is not realistic to expect our packs names match the git repos.
-James
On Tue, Sep 1, 2015 at 7:11 PM, Manas Kelshikar <notifications@github.com>
wrote:
I don't buy your bug :p

pack_name = repo_name

—
Reply to this email directly or view it on GitHub
[https://github.com/StackStorm/st2-kafka/pull/3#issuecomment-136898025] .[https://github.com/notifications/beacon/AABOPJ3TGAopKV83TKAcIqu5SFlDYxYaks5otjYXgaJpZM4F2Ea4.gif]

**Comment by armab** [comment]
```sh
# st2 run packs.install packs=st2-kafka branch=rework-branching repo_url=stackstorm/st2-kafka
# st2ctl reload --register-all
```

Output: 
```
Registering content...[flags = --register-all]
2015-09-02 00:24:09,725 INFO [-] Connecting to database "st2" @ "0.0.0.0:27017" as user "None".
2015-09-02 00:24:09,854 INFO [-] =========================================================
2015-09-02 00:24:09,855 INFO [-] ############## Registering sensors ######################
2015-09-02 00:24:09,857 INFO [-] =========================================================
2015-09-02 00:24:10,403 INFO [-] Registered 2 sensors.
2015-09-02 00:24:10,417 INFO [-] =========================================================
2015-09-02 00:24:10,418 INFO [-] ############## Registering actions ######################
2015-09-02 00:24:10,419 INFO [-] =========================================================
2015-09-02 00:24:15,942 INFO [-] Registered 58 actions.
2015-09-02 00:24:15,957 INFO [-] =========================================================
2015-09-02 00:24:15,958 INFO [-] ############## Registering rules ########################
2015-09-02 00:24:15,958 INFO [-] =========================================================
2015-09-02 00:24:16,284 ERROR [-] Failed registering rule from /opt/stackstorm/packs/st2-kafka/rules/parse_mesage.yaml.
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/st2reactor/bootstrap/rulesregistrar.py", line 107, in _register_rules_from_pack
    (pack, pack_field))
Exception: Model is in pack "st2-kafka" but field "pack" is different: kafka
2015-09-02 00:24:16,320 INFO [-] Registered 2 rules.
2015-09-02 00:24:16,326 INFO [-] =========================================================
2015-09-02 00:24:16,328 INFO [-] ############## Registering aliases ######################
2015-09-02 00:24:16,330 INFO [-] =========================================================
2015-09-02 00:24:16,765 INFO [-] Registered 11 aliases.
2015-09-02 00:24:16,769 INFO [-] =========================================================
2015-09-02 00:24:16,770 INFO [-] ############## Registering policy types #################
2015-09-02 00:24:16,772 INFO [-] =========================================================
2015-09-02 00:24:16,844 INFO [-] Registered 2 policy types.
2015-09-02 00:24:16,849 INFO [-] =========================================================
2015-09-02 00:24:16,849 INFO [-] ############## Registering policies #####################
2015-09-02 00:24:16,852 INFO [-] =========================================================
2015-09-02 00:24:16,855 INFO [-] Registered 0 policies.

```

Also +1, writing: `st2 run packs.install packs=st2-kafka` where pack name should be really `kafka` - confusing.

**Comment by manasdk** [comment]
I wrote my comment hastily and in the process did not explain my point of view properly.

I also agree that `repo_name` should not be `pack name`; I've run into this a few times myself. Although what we have right now is not ideal it is also not a bug it is simply how the system behaves and as such we should try to work within those limitations until we can fix the StackStorm platform. Not fixing the platform leads us down the path of organizing the repo this way which honestly is much more likely to stick as a pattern which will be even harder to undo. Imagine the scenario where there are 10s or 100s of packs that have been organized as stand-alone repos in the way that this pack structure is organized. At that point even if we support `repo_name != pack_name` we still have to deal with whatever fallout there is of the organization here.

Now I would further like to argue that the fix to support a `pack_name` which is distinct from `repo_name` should be made shortly. In the meanwhile is it completely attrocious to have `st2-kafka` or even rename this repo to `kafka`.

Anyhow, I have ranted enough and presented my point of view. Likely this discussion even takes away from the main purpose of this pack and work being done here anyway so I will not argue my point further.

TLDR; I fully agree that pack_name should not be repo_name however give the current state it is better to embrace that and fix the platform that to come up with workarounds.

**Comment by jfryman** [comment]
> TLDR; I fully agree that pack_name should not be repo_name however give the current state it is better to embrace that and fix the platform that to come up with workarounds.

:cool:. In principle we are in agreement. The only problem I had with fixing the real problem right now was that it required me to go and dig into the `download.py` in our download script which we always said we should revisit before :one:.:zero:.:zero:. Let's carve some :clock1: to revisit that script as a real workflow and get it fixed and then we can bring this PR back to life.

I agree that this is in fact an anti-pattern, and we need a solution. Meet me stage left, I actually have an idea about that too.

**Comment by armab** [comment]
@jfryman @manasdk  a bit of offtopic, but still about packs and naming.

Maybe it worth to create additional `Github organization` only for integration packs?
In current `stackstorm` org it's really hard to find what is pack and what's not. What will be when we'll have more and more "standalone" packs?

Most probably was discussed before, but I'll repeat obvious things. Advantages:
* easy to discover from `packs.install` side
* easy to search from user's side, where all org is about integrations
* nice and clear GitHub repo names, no ugly and unclear `st2-something`
* you know exactly how many integrations you have by navigating to org. url

Something like https://github.com/st2integrations/mysql or https://github.com/st2packs/slack looks nice.

**Comment by dzimine** [comment]
This comes as a "package", pun intended.

We need a "stackstorm forge" to pull all the packs together from multiple stackstorm and user repos.
Without it, spreading packs into individual repositories simply doesn't make much sense. 

I would much rather stick with the current model of st2contrib/st2incubator for the things WE do, until we figure the forge out.

Separate org as a stop-gap is in fact not a bad idea.

**Comment by jfryman** [comment]
Totally agree. Let's figure out how to make that work. 
https://stackstorm.slack.com/archives/stackstorm/p1441206067005207

**Comment by dennybaa** [comment]
+1 to org, the same way as salt guys made with their saltstack-formulas org. We've could already picked the approach a few months ago)

**Comment by dennybaa** [comment]
@dzimine, @jfryman, @armab  What do you mean by "stackstorm forge"
>  a "stackstorm forge" to pull all the packs together from multiple stackstorm and user repos.

I've seen two approaches so far:
 1. https://github.com/saltstack-formulas it's a separate org and it contains the whole "pack". Since where's a separate  repo it can be maintained not only "centrally", but also with help of community maintainers.
 1. Also there's another kind of an index like https://github.com/docker-library/official-images/tree/master/library. This one just basically contains "links" to a real place where a pack is located...

But it's clear that we should have some metadata in packs which can provide dependencies. Actually if we combine both, it could be nice.
Using the first method we get authority over pack repos, and indexing method can provide the input for dependency tracking.

**Comment by epowell101** [comment]
I like the way you frame alternatives.  

The vision that at least I was sold on was to make contributing to the community easier than not contributing.  As it is this is not the case; for example, folks fork us and hack on us and we loose track of their packs even if they keep them open.  

To address the above concerns one could argue for the Docker approach.

**Comment by dennybaa** [comment]
Yeah, that's exactly what I mean. Regarding the docker approach, it won't make us loose the track in case we point links into our `https://github.com/st2packs/some-pack` etc. So vital packs should be better offered through our org, and at the same time we can have links to other "non-vital" packs.

Docker approach seems nice, because it helps to maintain clear index in VCS without bringing many additional systems for the task. For example  https://supermarket.chef.io/ it's a big infra and indeed their cookbooks serve metadata, can be easily located and downloaded. However chef's way is a hard one to implement( We should come up with smth easy, but with a good usability.
